File: src/services/audit/detectors/ad/compliance/cis/network-security.ts
/**
 * CIS Network Security (2.3.7.x)
 * Checks CIS Benchmark network security settings
 */

import { Finding } from '../../../../../../types/finding.types';
import { GpoSecuritySettings } from '../../../../../../providers/smb/smb.provider';

export function detectCisNetworkSecurity(
  gpoSettings: GpoSecuritySettings | null,
  _includeDetails: boolean
): Finding {
  const issues: string[] = [];

  if (gpoSettings) {
    // Check SMBv1
    if (gpoSettings.smbv1ServerEnabled || gpoSettings.smbv1ClientEnabled) {
      issues.push('SMBv1 enabled (CIS 2.3.7.1 - Disable SMBv1)');
    }

    // Check LDAP signing
    if (gpoSettings.ldapServerIntegrity !== undefined && gpoSettings.ldapServerIntegrity < 2) {
      issues.push('LDAP signing not required (CIS 2.3.7.2)');
    }

    // Check LDAP channel binding
    if (gpoSettings.ldapChannelBinding !== undefined && gpoSettings.ldapChannelBinding < 2) {
      issues.push('LDAP channel binding not required (CIS 2.3.7.3)');
    }
  } else {
    issues.push('GPO settings not available for network security analysis');
  }

  return {
    type: 'CIS_NETWORK_SECURITY',
    severity: 'medium',
    category: 'compliance',
    title: 'CIS Benchmark - Network Security Non-Compliant',
    description:
      'Network security settings do not meet CIS Benchmark recommendations. SMBv1 should be disabled, LDAP signing required.',
    count: issues.length > 0 ? 1 : 0,
    details: issues.length > 0 ? { violations: issues, framework: 'CIS', control: '2.3.7.x' } : undefined,
  };
}

File: internal/audit/detectors/ad/compliance/cis/network-security.go
package cis

import (
	"context"

	"github.com/etcsec-com/etc-collector/internal/audit"
	"github.com/etcsec-com/etc-collector/pkg/types"
)

// NetworkSecurityDetector checks CIS network security compliance
type NetworkSecurityDetector struct {
	audit.BaseDetector
}

// NewNetworkSecurityDetector creates a new detector
func NewNetworkSecurityDetector() *NetworkSecurityDetector {
	return &NetworkSecurityDetector{
		BaseDetector: audit.NewBaseDetector("CIS_NETWORK_SECURITY", audit.CategoryCompliance),
	}
}

// Detect executes the detection
func (d *NetworkSecurityDetector) Detect(ctx context.Context, data *audit.DetectorData) []types.Finding {
	var issues []string

	// CIS 2.3.8.1: Microsoft network client: Digitally sign communications (always)
	// CIS 2.3.8.2: Microsoft network client: Digitally sign communications (if server agrees)
	// CIS 2.3.9.1: Microsoft network server: Digitally sign communications (always)
	// These settings would require GPO analysis - flag for manual review

	// Check LDAP signing via domain functional level hints
	if data.DomainInfo != nil {
		if data.DomainInfo.FunctionalLevelInt < 7 { // Windows Server 2016
			issues = append(issues, "CIS 2.3.11.8: Domain functional level may not enforce LDAP signing")
		}
	}

	finding := types.Finding{
		Type:        d.ID(),
		Severity:    types.SeverityMedium,
		Category:    string(d.Category()),
		Title:       "CIS Network Security Review Required",
		Description: "Network security settings require manual verification per CIS Benchmark. Check SMB signing, LDAP signing, and LDAP channel binding settings via GPO.",
		Count:       0,
		Details: map[string]interface{}{
			"framework": "CIS",
			"benchmark": "CIS Microsoft Windows Server Benchmark",
			"sections":  []string{"2.3.8 - Network client", "2.3.9 - Network server", "2.3.11 - Network security"},
			"note":      "GPO-based settings require manual verification",
		},
	}

	if len(issues) > 0 {
		finding.Count = 1
		finding.Details["violations"] = issues
	}

	return []types.Finding{finding}
}

func init() {
	audit.MustRegister(NewNetworkSecurityDetector())
}
